lib/ruby-mpd/parser.rb

```diff
@@ -24,15 +24,15 @@ def convert_command(command, *params)
             "#{param.begin}:#{param.end + (param.exclude_end? ? 0 : 1)}"
           end
         when MPD::Song
-          %Q["#{param.file}"] # escape filename
+          %Q["#{param.file.gsub('"','\\"')}"] # escape filename
```

Comment:
`Shellwords.escape` does not seem quite appropriate to me. From the documentation there:

> _Note that a resulted string should be used unquoted and is not intended for use in double quotes nor in single quotes._

MPD requires us to wrap in double-quotes; you can't just backslash escape characters.

Filename    | MPD requires    | Shellwords creates
------------|-----------------|-------------------
`foo.mp3`   | `foo.mp3`       | `foo.mp3`
`a b.mp3`   | `"a b.mp3"`     | `a\ b.mp3`
`bob's.mp3` | `"bob's.mp3"`   | `bob\'s.mp3`
`"yay".mp3` | `"\"yay\".mp3"` | `\"yay\".mp3`

We _could_ wrap the result of `Shellwords.escape` in double quotes and the results would be fine. However, all that does is add superfluous backslashes by calling `gsub` with a larger set of characters.

Would you prefer that I extract all escaping into a re-used utility method that is hand-tailored to MPD's needs? e.g.

```ruby
def mpd_argument(str)
   str =~ /['"\s]/ ? "#{str.gsub '"','\\"'}" : str
end
```

That way, if we find more bugs, more characters that need escaping, the fix is DRY.